src/recursive_sorting/recursive_sorting.py

```diff
@@ -1,33 +1,60 @@
 # TO-DO: complete the helpe function below to merge 2 sorted arrays
-def merge( arrA, arrB ):
-    elements = len( arrA ) + len( arrB )
+def merge(arrA, arrB):
+    elements = len(arrA) + len(arrB)
     merged_arr = [0] * elements
     # TO-DO
-    
+    i = j = k = 0
+
+    while i < len(arrA) and j < len(arrB):
+        if arrA[i] < arrB[j]:
+            merged_arr[k] = arrA[i]
+            i += 1
+        else:
+            merged_arr[k] = arrB[j]
+            j += 1
+        k += 1
+
+    while i < len(arrA):
+        merged_arr[k] = arrA[i]
+        i += 1
+        k += 1
+
+    while j < len(arrB):
+        merged_arr[k] = arrB[j]
+        j += 1
+        k += 1
+
     return merged_arr
 
 
 # TO-DO: implement the Merge Sort function below USING RECURSION
-def merge_sort( arr ):
+def merge_sort(arr):
     # TO-DO
-
+    midpoint = int(len(arr) / 2)
```

Comment:
setting a midpoint is good so you can use it when separating the two arrays. but this problem wants you to do this recursively which means you would want to be calling merge_sort for arr1 and arr2. and then using your helper function afterwards to put them together.

arr1 = merge_sort(arr[0:midpoint]
arr2 = merge_sort(arr[midpoint:]
arr = merge(arr1, arr2)
return arr